_posts/2025-05-22-introducing-uv.md

```diff
@@ -0,0 +1,311 @@
+- **Collaboration**: Team members and CI pipelines use the same environment.
+
+[PEP 751](https://peps.python.org/pep-0751/) introduces some standardization for Python lockfiles (`pylock.toml`) but many tools have not yet adopted this standard.
+`uv`'s lockfile data cannot yet be fully expressed in `pylock.toml` files and as a result it still depends on `uv.lock` files.
+It's like that the Python standard for lockfiles will evolve over time and could eventually converge (meaning you'd be able to use multiple tools with the same lockfile).
+
+### Processing code through `uv` environments
+
+```bash
+# run the boilerplate code through the uv environment
+# note: the `python -c` command flag executes inline code.
+uv run python -c "import project_name; project_name.main()"
+
+# run pytest through the uv environment
+uv run pytest
+```
+
+When we want to process code through the `uv` environment we can use the `uv run` command.
+If you're used to using `conda` environments this is akin to `conda run -n env_name python`.
+Note: `uv` does not enable you to activate or "enter into" an implicit shell for the environement like `conda activate`.
+Instead, `uv` uses declarative syntax to ensure the command-line interface to the environment is explicit.
+
+When working with other projects you might also need to run an `install` command in order to have access to the environment.
+`uv` skips this step and automatically will install the environment on issuing a `uv run` (there is no `uv install` command).
+
+When we use `uv run` several things happen:
+
+- `uv` will attempt to fetch a system or remote version of Python as stipulated in the `pyproject.toml` file (or make a guess at a version of Python to use).
+- `uv` will install or update an ephemeral virtual environment under a `~/.venv` directory (and use this environment to execute the process).
+
+### Building Python packages for distribution
+
+```bash
+# build a package for distribution
+uv build
+```
+
+`uv` provides a build system which enables you to build packages for distribution through the command `uv build`.
+By default it creates Pythonic `.whl` and `.tar.gz` formats that are common to PyPI.
+Note that you still have to upload these through other means in order for them to be hosted on common platforms like PyPI.
+We recommend using the [Trusted Publisher](https://docs.pypi.org/trusted-publishers/) method for publishing your packages once readied, which takes advantage of GitHub Actions or similar continuous integration / continuous deployment (CI/CD) tooling.
+
+```toml
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = [ "setuptools>=64", "setuptools-scm>=8" ]
+```
+
+Note that you can change your build backends within the `pyproject.toml` configuration (instead of using `uv-build` by default).
+For example, if you wanted to use `setuptools` you could stipulate something like the following in your `pyproject.toml` file instead.
```

Comment:
It seems like this sentence and the code sample might've changed positions during revision. I've updated the sentence, but you might consider moving the code sample below this paragraph instead.
```suggestion
For example, if you wanted to use `setuptools` you could stipulate something like the above in your `pyproject.toml` file instead.
```

Reply:
Thanks!